### PR TITLE
Added memoize for cats-effect

### PR DESCRIPTION
**Pull request by vitold**
Hi, I added memoize to the IOEffects, we plan to use it during our huge refactoring, so it would be awesome if you review the changes and if it's fine - release a new version. 
Thanks!

### CODE REVIEW

File: cats/shared/src/main/scala/org/atnos/eff/addon/cats/effect/IOEffect.scala [hunk 1]
```diff
@@ -72,6 +74,58 @@ trait IOInterpretation extends IOTypes {
...
+    interpret.translate(effect)(new Translate[Memoized, U] {
+      def apply[X](mx: Memoized[X]): Eff[U, X] =
+        mx match {
+          case Store(key, value) => IOEffect.ioDelay(cache.memo(key, value()))
+          case GetCache() => IOEffect.ioDelay(cache)
+        }
+    })
+  }
+
+  implicit val taskSequenceCached: SequenceCached[IO] = new SequenceCached[IO] {
```
Comment (etorreborre) [inline cats/shared/src/main/scala/org/atnos/eff/addon/cats/effect/IOEffect.scala]: This should be called `ioSequenceCached` isn't it?

**Comment by etorreborre** [comment]
Sorry for the late answer, please do the small suggestion and I will publish today.

File: cats/shared/src/main/scala/org/atnos/eff/addon/cats/effect/IOEffect.scala (hunk 1 above)
Comment (vitold) [reply cats/shared/src/main/scala/org/atnos/eff/addon/cats/effect/IOEffect.scala]: yep :)

**Comment by etorreborre** [comment]
Thanks!

**Comment by etorreborre** [comment]
I released `5.4.1`

**Pull request merged**